pandas/io/formats/format.py

```diff
@@ -1767,16 +1767,13 @@ def _format_datetime64_dateonly(
     nat_rep: str = "NaT",
     date_format: str | None = None,
 ) -> str:
-    if x is NaT:
+    if isinstance(x, NaTType):
```

Comment:
-1 on code changes to satisfy the type checker. We know that static type checker have issues with singletons and imo that was adequately noted in the removed comment adjacent to the ignore statement.

It may be nbd is some cases, but am always -1 on principle.